### PR TITLE
Add CircuitManagementPayload verification

### PR DESCRIPTION
**Pull request by shannynalayna**
Adds a method in the admin service to verify the CircuitManagementPayload fields, including the payload and header. Also adds signature verification in the `propose_change` method, where this was previously unchecked.

### CODE REVIEW

File: libsplinter/src/admin/shared.rs [hunk 1]
```diff
@@ -2066,7 +2069,220 @@ mod tests {
         if let Ok(_) =
             admin_shared.validate_circuit_vote(&vote, b"test_signer_a", &proposal, "node_a")
         {
-            panic!("Should have been invalid becasue the circuit hash does not match");
+            panic!("Should have been invalid because the circuit hash does not match");
+        }
+    }
+
+    #[test]
+    // test that the CircuitManagementPayload returns an error in case the signature is empty.
```
Comment (agunde406) [inline libsplinter/src/admin/shared.rs]: technically the CircuitManagementPayload does not return this error

File: libsplinter/src/admin/shared.rs [hunk 2]
```diff
@@ -514,24 +518,29 @@ impl AdminServiceShared {
...
-            Err(ServiceError::UnableToHandleMessage(_)) => (),
-            Err(err) => return Err(err),
-        };
-
         let header =
             protobuf::parse_from_bytes::<CircuitManagementPayload_Header>(payload.get_header())?;
+        self.validate_circuit_management_payload(&payload, &header)
+            .map_err(|err| ServiceError::UnableToHandleMessage(Box::new(err)))?;
+
+        self.verify_signature(&payload)?;
```
Comment (agunde406) [inline libsplinter/src/admin/shared.rs]: Either split this out into its own commit or add a message to the commit.

Comment (shannynalayna) [reply libsplinter/src/admin/shared.rs]: Split handling the unknown / unset actions into a separate commit and added a description of the signature verification to that existing commit.
